src/python/pants/backend/go/build_integration_test.py

```diff
@@ -35,116 +36,85 @@ def rule_runner() -> RuleRunner:
-    return built_package
-
-
-def test_package_one_target(rule_runner: RuleRunner) -> None:
+def test_package_simple(rule_runner: RuleRunner) -> None:
     rule_runner.write_files(
-        {"BUILD": "go_module(name='root')\n", "go.mod": "module foo.example.com\n"}
+        {
+            "go.mod": "module foo.example.com\n",
+            "BUILD": "go_module(name='root')\n",
+            "project/main.go": dedent(
+                """\
+                package main
+
+                import (
+                    "fmt"
+                )
+
+                func main() {
+                    fmt.Println("Hello world!")
+                }
+                """
+            ),
+            "project/BUILD": dedent(
+                """\
+                go_package(name='main', import_path='main')
+                go_binary(name='bin', binary_name='foo', main=':main')
+                """
+            ),
+        }
     )
-    target = make_target(rule_runner, [MAIN_SOURCE], import_path="main")
-    built_package = build_package(rule_runner, target)
+    binary_tgt = rule_runner.get_target(Address("project", target_name="bin"))
+    built_package = build_package(rule_runner, binary_tgt)
     assert len(built_package.artifacts) == 1
-    assert built_package.artifacts[0].relpath == "foo"
+    assert built_package.artifacts[0].relpath == "project/foo"
 
 
-def test_package_with_depenedency(rule_runner: RuleRunner) -> None:
+def test_package_with_dependency(rule_runner: RuleRunner) -> None:
     rule_runner.write_files(
-        {"BUILD": "go_module(name='root')\n", "go.mod": "module foo.example.com\n"}
-    )
-    lib_target = make_target(
-        rule_runner, [LIB_SOURCE], import_path="example.com/lib", target_name="lib"
-    )
-    main_target = make_target(
-        rule_runner,
-        [MAIN_SOURCE],
-        import_path="main",
-        dependencies=(lib_target.address,),
-        target_name="main",
+        {
+            "go.mod": "module foo.example.com\n",
+            "BUILD": "go_module(name='root')\n",
+            "project/lib/lib.go": dedent(
+                """\
+                package lib
+
+                import (
+                    "fmt"
+                )
+
+                func Quote(s string) string {
+                    return fmt.Sprintf(">> %s <<", s)
+                }
+                """
+            ),
+            "project/lib/BUILD": "go_package(import_path='example.com/lib')",
+            "project/main.go": dedent(
+                """\
+                package main
+
+                import (
+                    "fmt"
+                    "example.com/lib"
+                )
+
+                func main() {
+                    fmt.Println(lib.Quote("Hello world!"))
+                }
+                """
+            ),
+            "project/BUILD": dedent(
+                """\
+                go_package(name='main', import_path='main', dependencies=['project/lib'])
```

Comment:
No need to set `import_path` here. The go_binary rules will handle that properly.